src/lib/services/query-service.ts

```diff
@@ -0,0 +1,47 @@
+import type { WorkflowExecution } from '$lib/models/workflow-execution';
+import { requestFromAPI } from '$lib/utilities/request-from-api';
+
+type StackTraceOptions = {
+  workflow: WorkflowExecution;
+  namespace: string;
+};
+
+type QueryPayload = {
+  data: string;
+  metadata: {
+    encoding?: string;
+  };
+};
+
+type QueryType = {
+  payloads: QueryPayload[];
+};
+
+type StackTraceExecution = {
+  queryRejected?: null;
```

Comment:
So, this is optional, but when it's present, it's always null?

Reply:
fixed this one it wasn't optional it's either a string or null